src/mobile/src/ui/views/wallet/ViewSeed.js

```diff
@@ -151,7 +130,13 @@ class ViewSeed extends Component {
 
     componentWillReceiveProps(newProps) {
         if (this.props.seedIndex !== newProps.seedIndex) {
-            this.hideSeed();
+            this.resetComponent();
+        }
+    }
+
+    componentWillUpdate(newProps, newState) {
+        if (this.state.token.length !== 6 && newState.token.length === 6) {
```

Comment:
I think we should avoid magic numbers and should have a symbolic constant for token length.